ytypes/leaf_test.go

```diff
@@ -1965,6 +1997,42 @@ func TestUnmarshalLeafGNMIEncoding(t *testing.T) {
 		},
 		{
 			desc:     "success unmarshalling union leaf string field",
+			inSchema: unionSchemaSimple,
+			inVal: &gpb.TypedValue{
+				Value: &gpb.TypedValue_StringVal{
+					StringVal: "forty two",
+				},
+			},
+			wantVal: &LeafContainerStruct{UnionLeafSimple: testutil.String("forty two")},
```

Comment:
Do we support this for the 'complex' union types -- or is this just supported for the simple type?

Reply:
You mean this test case?
https://github.com/openconfig/ygot/blob/40495a5034a44232bb28b0a4ef8e40d20e694432/ytypes/leaf_test.go#L2034-L2043

Reply:
Ah, yes, apologies, I'd missed this one.